src/core/parser/project.ts

```diff
@@ -0,0 +1,30 @@
+import { parseComponentJSON } from './component'
+
+import { ProjectUIDL, ComponentUIDL } from '../../typings/uidl-definitions'
+import { cloneObject } from '../../shared/utils/uidl-utils'
+
+interface ParseProjectJSONParams {
+  noClone?: boolean
+}
+
+export const parseProjectJSON = (
+  input: Record<string, unknown>,
+  params: ParseProjectJSONParams = {}
+): ProjectUIDL => {
+  const safeInput = params.noClone ? input : cloneObject(input)
+  const root = safeInput.root as ComponentUIDL
+
+  const result = {
+    ...(safeInput as ProjectUIDL),
+  }
+
+  result.root = parseComponentJSON(root)
```

Comment:
will you pass `noClone` as a param at this point?

Reply:
good point